File: Python3/898. Bitwise ORs of Subarrays.py
class Solution:
    def subarrayBitwiseORs(self, arr: List[int]) -> int:
        prev_set = set()
        glob_set = set()

        for n in arr:
            curr_set = {n}
            for p in prev_set:
                curr_set.add(n | p)

            glob_set |= curr_set
            prev_set = curr_set

        return len(glob_set)

File: Java/898. Bitwise ORs of Subarrays.java
class Solution {
    public int subarrayBitwiseORs(int[] arr) {
        Set<Integer> prev = new HashSet<>();
        Set<Integer> glob = new HashSet<>();

        for (int n: arr) {
            Set<Integer> curr = new HashSet<>();
            curr.add(n);
            for (int p: prev) {
                curr.add(n | p);
            }
            glob.addAll(curr);
            prev = curr;
        }
        
        return glob.size();
    }
}